src/libs/auth.ts

```diff
@@ -97,71 +127,193 @@ const revokeTokens = async () => {
+            // Auth token expiration is set to 24 hours in B2C configuration.
+            expiresAt: authTokenCreatedAt + 86400, // 24 hours in seconds
+            totalTokensUsedThisSession: authStore.get().authTokenMetadata.totalTokensUsedThisSession + 1,
+            totalTokenLoadAttemptsThisSession: authStore.get().authTokenMetadata.totalTokenLoadAttemptsThisSession + 1,
+          }
+        : { ...authStore.get().refreshTokenMetadata },
+    }));
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.success, {
+      authProvider: (user.profile as B2cIdTokenClaims).idp,
+      ...getOldAuthTokenLabels(oldAuthTokenMetadata),
+      authTokenCreatedAt: getTimestampMetricLabel(authTokenCreatedAt),
+      authTokenExpiresAt: getTimestampMetricLabel(authTokenExpiresAt),
+      authTokenId,
+      refreshTokenId: authStore.get().refreshTokenMetadata.id,
+      refreshTokenCreatedAt: getTimestampMetricLabel(authStore.get().refreshTokenMetadata.createdAt),
+      refreshTokenExpiresAt: getTimestampMetricLabel(authStore.get().refreshTokenMetadata.expiresAt),
+      jwtExpiresAt: getTimestampMetricLabel(jwtExpiresAt),
+    });
+  } else if (loadedAuthTokenState.status === 'expired') {
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.expired, {
+      ...getOldAuthTokenLabels(oldAuthTokenMetadata),
+      refreshTokenId: authStore.get().refreshTokenMetadata.id,
+      refreshTokenCreatedAt: getTimestampMetricLabel(authStore.get().refreshTokenMetadata.createdAt),
+      refreshTokenExpiresAt: getTimestampMetricLabel(authStore.get().refreshTokenMetadata.expiresAt),
+    });
+  } else {
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.error, {
+      // we could potentially log the reason, but I don't know if that data is safe to log
+      ...getOldAuthTokenLabels(oldAuthTokenMetadata),
+      refreshTokenId: authStore.get().refreshTokenMetadata.id,
+      refreshTokenCreatedAt: getTimestampMetricLabel(authStore.get().refreshTokenMetadata.createdAt),
+      refreshTokenExpiresAt: getTimestampMetricLabel(authStore.get().refreshTokenMetadata.expiresAt),
+    });
+  }
+  return loadedAuthTokenState;
+};
+
+const tryLoadAuthToken = async (includeBillingScope = false, popUp = false): Promise<AuthTokenState> => {
+  const args: ExtraSigninRequestArgs = getSigninArgs(includeBillingScope);
+  const authInstance = getAuthInstance();
+  const loadedAuthTokenResponse: User | null = await (popUp
+    ? // returns Promise<User | null>, attempts to use the refresh token to get a new authToken
+      authInstance.signinPopup(args)
+    : authInstance.signinSilent(args)
+  ).catch((e) => {
+    const userErrorMsg = 'Unable to sign in.';
+    const internalErrorMsg = 'An unexpected exception occurred while attempting to load new auth token.';
+    console.error(`user error message: ${userErrorMsg}`);
+    console.error(`internal error message: ${internalErrorMsg}`);
+    console.error(`reason: ${e}`);
```

Comment:
Nit: suggest simplifying this message.

```suggestion
    console.error('Unable to sign in due to error reloading access token', e);
```